### DISCUSSION TITLE
Dear Andrés M... Let's reorganize.

### DISCUSSION THREAD

**Discussion by mipmip** [andmarti1424/sc-im#504] [General]
Dear Andrés M.,

You prevented Sc from being forgotten and added great features to the reborn Sc-im. First time I visited your project it had 1K stars, since than the stars have doubled. A lot of people love this unique piece of good old fashioned unix software. Unfortnately the project is standing still and PR's stay open. 

You seem to be too busy to invest time into this one-man-project and that's not good for Sc-im. Sc-im deserves to be actively developed by a larger community of enthousiast volunteers. Vim and Neovim are more popular than ever, so now is the moment for Sc-im to grow it's own community. 

I'd like to invite you and everyone who is interested to discuss a new form of collaboration so that more people can contribute and the development goes faster. I know a lot of experienced developers would like help or be part of such a project.

Kind regards,
Pim Snel

**Comment by andmarti1424** [comment]
Yes. That's a good idea. Hope there is more people like you interested in collaborate. I am in

**Comment by mipmip** [comment]
Wow that's a quick reply. Great to hear you like the idea. :+1:

**Comment by mipmip** [comment]
As you are the founder of this project, your thoughts on how this project should be lead are now most important. Do you have idea's how you think it can be a larger community. And do you have a long term vision on how sc-im should evolve? This thoughts could be part of a mission statement.

**Comment by andmarti1424** [comment]
Well that's the thing. I do not know how it to become bigger. Do you have any ideas? About scim itself? I think the best would be to give more support.

**Comment by mipmip** [comment]
There might be templates to convert a personal project into a larger community project. I volunteered  in a few projects but I'm not manager myself :)

This might be a could start: https://opensource.guide/starting-a-project/ from section 3.

**Comment by mipmip** [comment]
I think a good start is too claim an organization name. sc-im is still free, so go on and claim it :) Within an organization it is easy to add volunteers you trust.

**Comment by msharov** [comment]
You don't need an organization to add volunteers; anybody can already be given write access to the project if that's what's necessary. Neither will volunteers magically appear if there is an "organization" attached. As things stand now, the main issue is more with lack of interest in sc rather than any difficulty in working on it, and that is not a problem we can solve.

**Comment by YAMLcase** [comment]
I have to lend my support to this idea!  I have been on the hunt for a bare-bones, no frills, no formatting spreadsheet app for years and sc-im is the ONLY one available.  Bonus (for me) is that it is linux CLI.  Thanks @mipmip for the "starting a project" link, this looks like a really good checklist.

**Comment by poetaman** [comment]
@msharov I think there will be immense interest, I didn't even know if such a thing existed in past. As a programmer, I had never been in love making new spreadsheets before trying sc-im, a CLI spreadsheet program.

We software developers, and nerds forget at times that tools are a human-interface. I think more improvements will only invite more people over time, this program has great potential like vim/neovim. I  started working on a [Dracula](https://draculatheme.com) theme for sc-im last night. 

@mipmip @andmarti1424 Given I don't see any commits since Sep 4 & there are show-stopper bugs (like https://github.com/andmarti1424/sc-im/issues/497), I have posted an invitation for enthusiastic developers to chip in on ycombinator: https://news.ycombinator.com/item?id=26260201 and also posted on neovim (given there is so much commonality with the interface of this program): https://github.com/neovim/neovim/issues/14011. Am not trained in C, so can't dive directly into it myself. Though I can help sponsor this project, and contribute in other ways.

PS: Sample dracula color scheme for sc-im. Am still refining it. Though having an ability to use more colors would definitely be helpful, I have opened another ticket for that: https://github.com/andmarti1424/sc-im/issues/495
<img width="1043" alt="109081692-8527c180-76b7-11eb-9229-1ee53066ea01" src="https://user-images.githubusercontent.com/71736629/109111529-f1251c80-76ed-11eb-96c8-9e1e8fa22e39.png">

**Comment by andmarti1424** [comment]
@reportaman That would be great. Hope more people chime in so this piece of software move to another level.

**Comment by Iron-E** [comment]
> also posted on neovim (given there is so much commonality with the interface of this program)

I could see the benefit of using Neovim as a backend. For one, it could potentially allow plugin creation for this application using Lua(!!) which is a big bonus.

I know that Neovim supports creating external UIs on top of it (see `:help ui` for more info) but I am not sure if it is feasible to create a tabular UI which calls back to Neovim. I am completely unversed in C though, so I don't really have much to offer in terms of concrete solutions.

> Hope more people chime in so this piece of software move to another level.

As someone who uses this program pretty frequently, I think a big thing that would help propel it forward is cross-compatibility with :"industry standard" tabular editors such as Excel. If we can somehow get our `.sc` files to export with formulas intact (kind of hit-or-miss at the moment) that would go a long way towards increasing usability. It's a shame monolithic XML zips have taken over, `.sc` is a nice format (there might be some benefit to `.xlsx` I'm not aware of though).

**Comment by andmarti1424** [comment]
With LUA you can make read and write calls to SC-IM. This could be the communication to other applications.
But again, I would replace the core for something new (remember the evaluation core is from the 80's!)
Something that supports multiple sheets and cross references would be necessary.

About Excel, the thing is in sc-im there are not the same formulas as used on Excel. Nevertheless, the main 8 functions should be exported to .xlsx files without issues (see line 628 of xlsx.c)..

**Comment by DarkWiiPlayer** [comment]
@andmarti1424 [Lua](http://lua.org/about.html#name)*

**Comment by Iron-E** [comment]
> With LUA you can make read and write calls to SC-IM. This could be the communication to other applications.

I must have forgotten it can already do this. Perhaps making a Neovim plugin for `sc-im` would actually be more readily doable than I originally thought.

I left a mock on the Neovim issue of something that would be cool to do (rendering tables with `sc-im`). It'll take me some time to investigate but it would be worth trying IMO.

What you said about reworking the core is probably a good idea if it's that old. Wish I could help more with C but if there were to be a significant rewrite I would advocate for Rust. Again that's all from the perspective of someone who doesn't know the internals or dependencies

**Comment by poetaman** [comment]
@andmarti1424 Add a GitHub sponsorship button to your repo!

**Comment by andmarti1424** [comment]
@reportaman I just dont know which funding model to use..
Patreon may be?

**Comment by poetaman** [comment]
@andmarti1424 I have been following the project Asahi Linux, this is what its contributions page says: https://asahilinux.org/support/. They use both Patreon, and GitHub sponsorship. Given your GitHub page already has 2.6k likes, it is a good place to begin I think, you can create both together.

**Comment by andmarti1424** [comment]
I have created a patreon. I still have to see how much are the fees. For instance, Paypal directly, cuts 5.7% in my case. which is a lot!

**Reply by poetaman** [reply]
Apparently GitHub doesn't cut anything, 100% goes to developers: https://twitter.com/github/status/1131477509156085760

**Comment by unrelentingfox** [comment]
It's been years since I programmed in C or C++ but I would love to offer assistance in any way I can, I'm currently working full time as a Java developer so I'm sure I could get back into the swing of using C. I absolutely love sc-im, it is filling the need for a no-frills, keyboard friendly alternative to excel. There are basically no other options that even come close. I'd love to see this project continue to grow.

**Comment by andmarti1424** [comment]
@unrelentingfox that would be great. thanks. I fixed a bunch of bugs in the last couple of months and sure added a bunch more with the added features in the middle. I will be opening new discussions for specific topic such as how the Evaluation should be made.
I also think that as soon as more volunteers join, it would be good idea to start a new core supporting multiple sheets.

**Reply by 1dancook** [reply]
I have found some other bugs in 0.8.2_2. Would you like me to work on reporting them? 
At least, what I can try is see if they have persisted in the dev branch. I suppose it is good to have the issue on GitHub and marked as "fixed" just so that others are aware if they come looking. 

I don't know how to code in C but I am pretty good at finding bugs and things that aren't quite right. 😜

I want to take the deep dive and start using sc-im for my daily spreadsheet use. It's really awesome. So hopefully I can help by giving feedback and reporting issues etc. 

I took a look at sc-im a long while back - version 0.7. At that point I think I couldn't keep using it because of crashing. But the project has matured 👍👍💪💪 and certainly ironing out the big bugs will help people stay on board as users.

**Reply by andmarti1424** [reply]
Yes please. THanks.

**Comment by andmarti1424** [comment]
@unrelentingfox btw I need help with something. Could you please write me to scim.spreadsheet@gmail.com ? Thanks

**Comment by oaklandgit** [comment]
I'm not a C programmer (yet) but I want to support this project and see it grow. Just became a Patron.
